http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/dao/google/GoogleOAuth2Service.scala

```diff
@@ -0,0 +1,30 @@
+package org.broadinstitute.dsde.workbench.leonardo
+package dao
+package google
+
+import cats.effect.concurrent.Semaphore
+import cats.effect.{Async, Blocker, ContextShift, Resource, Sync, Timer}
+import cats.mtl.ApplicativeAsk
+import com.google.api.client.googleapis.javanet.GoogleNetHttpTransport
+import com.google.api.client.json.jackson2.JacksonFactory
+import com.google.api.services.oauth2.Oauth2
+import io.chrisdavenport.log4cats.StructuredLogger
+import org.broadinstitute.dsde.workbench.model.{TraceId, UserInfo}
+
+trait GoogleOAuth2Service[F[_]] {
```

Comment:
I thought about putting `GoogleOAuth2Service`/`GoogleOAuth2Interpreter` in `workbench-libs/google2`, but left it in Leo for now. I could move if people feel strongly.